source/headers.ts

```diff
@@ -107,6 +143,36 @@ export const setDraft7Headers = (
 	)
 }
 
+/**
+ * Sets `RateLimit` & `RateLimit-Policy` headers based on the eighth draft of the spec.
+ * See https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-ratelimit-headers-08.
+ *
+ * @param response {Response} - The express response object to set headers on.
+ * @param info {RateLimitInfo} - The rate limit info, used to set the headers.
+ * @param windowMs {number} - The window length.
+ * @param key {string} - The unique string identifying the client.
+ */
+export const setDraft8Headers = (
+	response: Response,
+	info: RateLimitInfo,
+	windowMs: number,
+	key: string,
+): void => {
+	if (response.headersSent) return
+
+	const windowSeconds = Math.ceil(windowMs / 1000)
+	const resetSeconds = getResetSeconds(info.resetTime, windowMs)
+	const duration = getDurationInWords(windowMs)
+	const partitionKey = getPartitionKey(key)
+
+	const name = `rl-${info.limit}-in-${duration}`
```

Comment:
I like this as a default, but I think we should provide an option to customize the name.

Reply:
How would we accept the custom name? We could probably resurrect the `header` option we had deprecated, and make it accept a an object like so:

```js
const options = {
	// ...
	headers: {
		legacy: false,
		standard: {
			version: 'draft-8',
			name: 'burst',
			partition: 'storage'
		}
	}
}
```

This would also let us allow the user to generate their own `partitionKey`.

Or, we could add a new parameter called `name` that acts as the more generic name of the rate limiter, and the partition key remains auto-generated by us.

I'm in favour of this option for now since it is much simpler and does not involve a possibly breaking change, but the first option might be better in the long term if the drafts keep changing and adding more parameters.

Reply:
I was thinking of a new option like `identifier`, or `headersIdentifier`, or similar. `name` works too. It would just be ignored unless draft8+ is used. I don't think we should resurrect the `headers` option until the standard is completed.

Reply:
Okay, will use `identifier` then.